lckernel/cad/math/intersect.cpp

```diff
@@ -270,9 +250,22 @@ bool Intersect::operator()(const lc::entity::Ellipse& e, const lc::entity::Splin
     return false;
 }
 
-bool Intersect::operator()(const lc::entity::Ellipse& e, const lc::entity::LWPolyline& lwp) {
-    std::cerr << __PRETTY_FUNCTION__ << " requires implementation" << std::endl;
-    return false;
+bool Intersect::operator()(const lc::entity::Ellipse& a, const lc::entity::LWPolyline& lwp) {
+    auto &list1 = lwp.asEntities();
+    // Note: The dynamic_pointer_cast won't win a beauty contest, but the plan is to split
+    // the EntityVisitor into a GeoVisitor and EntityVisitor such that a application deciding
+    // to use double dispatch can decide to use a specific implementation.
+    // Once added, we can get rid of the dynamic_pointer_casts and simply
+    // call entity1.visit(entity2);
+    for (auto &entity1 : list1) {
+        if (auto arc = std::dynamic_pointer_cast<const lc::geo::Arc>(entity1)) {
+            geovisit(a, *arc.get());
+        }
+        else {
+            geovisit(*std::dynamic_pointer_cast<const lc::geo::Vector>(entity1).get(), a);
+        }
+    }
```

Comment:
I like this strategy. However, if you want to get rid of the `dynamic_pointer_casts`, you should, for each entity of the polyline, create an intersection and add the result to the intersection points.
In this case, you can change this function to take a `LWPolyline` and a `Visitable`.